File: my-sql-context/src/test/scala/vip/chengchao/LoggerTest.scala
package vip.chengchao

import org.junit.Test
import org.slf4j.LoggerFactory

class LoggerTest {


  val logger = LoggerFactory.getLogger("LoggerTest")

  @Test
  def printLogTest() :Unit = {

    logger.error("hello error")
    logger.warn("hello warn")
    logger.info("hello info")
    logger.debug("hello debug")
    logger.trace("hello trace")

    println("== end ==")
  }

  @Test
  def printAcsii(): Unit = {

    val longVal = 32L
    val intVal = longVal.asInstanceOf[Int]

    val charVal = intVal.asInstanceOf[Char]


    val c:Char = 65

    println("c == "+ c)

    for (i <- 0 to 93) {
      logger.info("i: {} = {}",i, (i + 33).asInstanceOf[Char])
    }
  }
}

File: group-netty/mock-gb-server/src/test/java/luxe/chaos/netty/mock/gb/other/LoggerTest.java
package luxe.chaos.netty.mock.gb.other;

import org.junit.Test;
import org.slf4j.*;

public class LoggerTest {

    private static final Logger logger = LoggerFactory.getLogger(LoggerTest.class);
    private static final Marker MARKER = MarkerFactory.getMarker("test_marker");

    @Test
    public void testMarker() {


        MDC.put("first", "Dorothy");
        String info = "hello.";

        logger.info(MARKER, "has marker ... she say : {}", info);
        logger.info("no marker ... she say : {}", info);

        // The most beautiful two words in the English language according
        // to Dorothy Parker:
        logger.info("Check enclosed.");
        logger.debug("The most beautiful two words in English.");

        MDC.put("first", "Richard");
        MDC.put("last", "Nixon");
        logger.info("I am not a crook.");
        logger.info("Attributed to the former US president. 17 Nov 1973.");
    }
}
